operations/_scripts/generate/generate_app_repo.sh

```diff
@@ -11,11 +11,15 @@ mkdir -p "${GITHUB_ACTION_PATH}/operations/deployment/ansible/app/${GITHUB_REPO_
 
 TARGET_PATH="$GITHUB_WORKSPACE"
 if [ -n "$APP_DIRECTORY" ]; then
-    echo "APP_DIRECTORY: $APP_DIRECTORY"
-    TARGET_PATH="${TARGET_PATH}/${APP_DIRECTORY}"
+  echo "APP_DIRECTORY: $APP_DIRECTORY"
+  TARGET_PATH="${TARGET_PATH}/${APP_DIRECTORY}"
 fi
 
-cp -rf "$TARGET_PATH"/* "${GITHUB_ACTION_PATH}/operations/deployment/ansible/app/${GITHUB_REPO_NAME}/"
+if [ -f "$TARGET_PATH/.ignore" ]; then
+  rsync -a --exclude-from="$TARGET_PATH/.ignore" "$TARGET_PATH"/ "${GITHUB_ACTION_PATH}/operations/deployment/$1/app/${GITHUB_REPO_NAME}/"
+else
+  rsync -a "$TARGET_PATH"/ "${GITHUB_ACTION_PATH}/operations/deployment/$1/app/${GITHUB_REPO_NAME}/"
+fi
```

Comment:
since i'm a stickler for keeping things DRY, i might rewrite this as:
```
rsync_cmd='"$TARGET_PATH"/ "${GITHUB_ACTION_PATH}/operations/deployment/$1/app/${GITHUB_REPO_NAME}/"'

if [ -f "$TARGET_PATH/.ignore" ]; then
	rsync_cmd="--exclude-from=\"$TARGET_PATH/.ignore\" $rsync_cmd"
fi

rsync -a $rsync_cmd
```

Reply:
I really like it, but it's a little bit harder to read for a newcomer. 
As it's the same amount of lines ... not sure if it's worthy. 
@mickmcgrath13 any thoughts?